creator-node/src/routes/nodeSync.js

```diff
@@ -271,121 +298,67 @@ async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint, dbOnlySync
-            // the transaction declared outside the try/catch needs to be closed. if we call the continue
-            // and do not end the tx, it will never be closed
-            transaction.rollback()
-            continue
-          }
-
-          const cnodeUserUUID = cnodeUser.cnodeUserUUID
-          req.logger.info(redisKey, `beginning delete ops for cnodeUserUUID ${cnodeUserUUID}`)
-
-          const numAudiusUsersDeleted = await models.AudiusUser.destroy({
-            where: { cnodeUserUUID },
-            transaction
-          })
-          req.logger.info(redisKey, `numAudiusUsersDeleted ${numAudiusUsersDeleted}`)
-
-          // TrackFiles must be deleted before associated Tracks can be deleted.
-          const numTrackFilesDeleted = await models.File.destroy({
-            where: {
-              cnodeUserUUID,
-              trackBlockchainId: { [models.Sequelize.Op.ne]: null } // Op.ne = notequal
-            },
-            transaction
-          })
-          req.logger.info(redisKey, `numTrackFilesDeleted ${numTrackFilesDeleted}`)
-
-          const numTracksDeleted = await models.Track.destroy({
-            where: { cnodeUserUUID },
-            transaction
-          })
-          req.logger.info(redisKey, `numTracksDeleted ${numTracksDeleted}`)
-
-          // Delete all remaining files (image / metadata files).
-          const numNonTrackFilesDeleted = await models.File.destroy({
-            where: { cnodeUserUUID },
-            transaction
-          })
-          req.logger.info(redisKey, `numNonTrackFilesDeleted ${numNonTrackFilesDeleted}`)
-
-          const numClockRecordsDeleted = await models.ClockRecord.destroy({
-            where: { cnodeUserUUID },
-            transaction
-          })
-          req.logger.info(redisKey, `numClockRecordsDeleted ${numClockRecordsDeleted}`)
-
-          const numSessionTokensDeleted = await models.SessionToken.destroy({
-            where: { cnodeUserUUID },
-            transaction
-          })
-          req.logger.info(redisKey, `numSessionTokensDeleted ${numSessionTokensDeleted}`)
-
-          // Delete cnodeUser entry
-          await cnodeUser.destroy({ transaction })
-          req.logger.info(redisKey, `deleted cnodeUserEntry`)
-        }
-
-        /* Populate all new data for fetched cnodeUser. */
-
         req.logger.info(redisKey, `beginning add ops for cnodeUserUUID ${fetchedCnodeUserUUID}`)
 
-        // Upsert cnodeUser row.
-        await models.CNodeUser.create({
-          cnodeUserUUID: fetchedCnodeUserUUID,
-          walletPublicKey: fetchedWalletPublicKey,
-          latestBlockNumber: fetchedLatestBlockNumber,
-          lastLogin: fetchedCNodeUser.lastLogin,
-          clock: fetchedCNodeUser.clock,
-          createdAt: fetchedCNodeUser.createdAt
-        }, { transaction })
-        req.logger.info(redisKey, `Inserted nodeUser for cnodeUserUUID ${fetchedCnodeUserUUID}`)
+        // Upsert cnodeUser row
+        await models.CNodeUser.upsert(
```

Comment:
What if there's two users with the same wallet under two cnodeUserUUID's, and there's db records for the different cnodeUserUUID in their respective db's?

Reply:
i just want to make sure we can reconcile to one user with the correct state. like basically you query by wallet above and then upsert the cnodeUserUUID and then use that everywhere else. like in the case where the cnodeUserUUID is different for some reason, you'd just add a new row in the CNodeUsers table, and then write the clockrecords and files starting at some paginated value as opposed to from the start like we used to do.

Reply:
discussed offline - this is a good catch. sync has been changed to always use local uuid if present

Reply:
added test for this condition